### PR TITLE
[CI] fail fast stages

### PR DESCRIPTION
**Pull request by v1v**
We got some lack of resources so let's fail fast to be able to launch some other builds.

### CODE REVIEW

**Review by mdelapenya — APPROVED** [review]

**Comment by mdelapenya** [comment]
I think we use same strategy in the e2e, right?

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -46,7 +46,7 @@ pipeline {
       }
     }
     stage('Test Hosts'){
-      failFast false
+      failFast true
```
Comment (mdelapenya) [inline .ci/Jenkinsfile:49]: wdyt about using `parallelsAlwaysFailFast()` in the top-level `options` section? Any preference? I see it more readable being declared at the beginning of the pipeline

Comment (v1v) [reply .ci/Jenkinsfile:49]: No strong opinion, but I've never used `parallelsAlwaysFailFast()` but failFast, so I can change it

Comment (mdelapenya) [reply .ci/Jenkinsfile:49]: I ended up using failFast (as in this PR) in https://github.com/elastic/e2e-testing/pull/429 See there rationale

**Review by cachedout — APPROVED** [review]

**Pull request merged**